src/expr/src/scalar/func.rs

```diff
@@ -367,18 +367,22 @@ fn cast_numeric_to_int64<'a>(a: Datum<'a>) -> Result<Datum<'a>, EvalError> {
 
 fn cast_numeric_to_float32<'a>(a: Datum<'a>) -> Result<Datum<'a>, EvalError> {
     let a = a.unwrap_numeric().0;
-    let mut cx = numeric::cx_datum();
-    match cx.try_into_f32(a) {
-        Ok(i) => Ok(Datum::from(i)),
-        Err(_) => Err(EvalError::Float32OutOfRange),
+    let i = a.to_string().parse::<f32>().unwrap();
+    if i.is_infinite() {
```

Comment:
Why is this needed? In postgres 14 you can do:

```
postgres=# select 'inf'::numeric::float4;
  float4  
----------
 Infinity
```

Reply:
Oh, interesting--I didn't realize `'inf'::numeric` is valid as of PG 14. Our current numeric type aims for PG 13 semantics, so the code in question passes must in that regar––but I filed #7427 to track the work here.

I'm going to merge this as-is primarily because I'm OOO until next Tuesday, want to unbreak SLT, and will need a few days to implement PG 14-style infinity.